core/src/map_constructor.rs

```diff
@@ -0,0 +1,7 @@
+use crate::editor::MapManager;
+
+pub mod shiftnanigans;
+
+pub trait MapConstructor {
+    fn construct_map(&self, map_manager: &mut MapManager);
```

Comment:
I like the idea of making a trait out of this. I could definitely see having different map generation techniques eventually. For example, something like [mario-gpt](https://github.com/shyamsn97/mario-gpt) has already been brought up.

Reply:
That was exactly what I was thinking too, which is why I was wanting to ensure that the `MapManager` could handle everything necessary for working with tiles/elements in the editor. I'm so glad that it's finally done and working well. I'll take a swing at that opened issue in Shiftnanigans next and then I'll work through the occasional freezes after that. Thank you for all of your help. I really appreciate it.

Reply:
I'm so glad I can help! Thank you for improving our game! :heart: